packages/shared/src/common/file-type.ts

```diff
@@ -0,0 +1,135 @@
+const TIFF_MAGIC_NUMBER_1 = 0x49;
+const TIFF_MAGIC_NUMBER_2 = 0x49;
+const TIFF_MAGIC_NUMBER_3 = 0x2a;
+const TIFF_MAGIC_NUMBER_4 = 0x00;
+const TIFF_MAGIC_NUMBER_5 = 0x4d;
+const TIFF_MAGIC_NUMBER_6 = 0x4d;
+const TIFF_MAGIC_NUMBER_7 = 0x00;
+const TIFF_MAGIC_NUMBER_8 = 0x2a;
+const PDF_MAGIC_NUMBER_1 = 0x25;
+const PDF_MAGIC_NUMBER_2 = 0x50;
+const PDF_MAGIC_NUMBER_3 = 0x44;
+const PDF_MAGIC_NUMBER_4 = 0x46;
+const PDF_MAGIC_NUMBER_5 = 0x2d;
+const XML_MAGIC_NUMBER_1 = 0x3c;
+const XML_MAGIC_NUMBER_2 = 0x3f;
+const XML_MAGIC_NUMBER_3 = 0x78;
+const XML_MAGIC_NUMBER_4 = 0x6d;
+const XML_MAGIC_NUMBER_5 = 0x6c;
+const XML_MAGIC_NUMBER_6 = 0x20;
+const PNG_MAGIC_NUMBER_1 = 0x89;
+const PNG_MAGIC_NUMBER_2 = 0x50;
+const PNG_MAGIC_NUMBER_3 = 0x4e;
+const PNG_MAGIC_NUMBER_4 = 0x47;
+const JPEG_MAGIC_NUMBER_1 = 0xff;
+const JPEG_MAGIC_NUMBER_2 = 0xd8;
+const BMP_MAGIC_NUMBER_1 = 0x42;
+const BMP_MAGIC_NUMBER_2 = 0x4d;
+
+function isLikelyTextFile(fileBuffer: Buffer): boolean {
```

Comment:
as discussed with Raf, uses likely ascii chars for text/plain type determination